+exp/SignalsExpTest.m

```diff
@@ -234,19 +246,19 @@ function newLayerValues(obj, name, val)
     function createLivePlot(obj)
     % creates the 'LivePlot' figure, if user-specified via GUI
     
-      if obj.ExpTest.LivePlot
-        obj.ExpTest.LivePlotFig = figure('Name', 'LivePlot',...
+      if obj.ETest.LivePlot
+        obj.ETest.LivePlotFig = figure('Name', 'LivePlot',...
           'NumberTitle', 'off', 'Color', 'w',...
           'DeleteFcn', @(~,~) emptyHandle(obj));
         sig.timeplot(obj.T, obj.Events,...
-          'Parent', obj.ExpTest.LivePlotFig, 'xlabelOff', 1);
+          'Parent', obj.ETest.LivePlotFig, 'xlabelOff', 1);
```

Comment:
Not sure when timeplot was changed but I would prefer that any axes properties be set on the returned handles and not implemented as an input arg.  The reason being that you either implement none or implement all.  It's inconsistent otherwise, e.g. if the user wants to set the xticks, they can't use `sig.timeplot(t, 'xticks', [])`.  

Instead, you could remove all input args pertaining to the axes handles and set them like this:
```
ax_h = sig.timeplot(obj.T, obj.Events, 'Parent', obj.ExpTest.LivePlotFig);
set(ax_h, 'XTickLabel', []);
````

This may also be quicker as they're all set at once, rather than one at a time.  Plus you don't have all that overhead in sig.timeplot for parsing these sorts of axes property arguments.  As line handles aren't directly returned, it would make more sense to allow those props to be set via input name-value pairs, as was previously the case.

Reply:
ok, when we talked about it before we agreed on the 'xlabelOff' flag, but sure your argument here makes more sense.

Reply:
Sorry to yo-yo on things, I know it's annoying when people do that haha

Reply:
Eh it's fine I wouldn't even call this a yo-yo, because you're suggesting a new idea. Also I'm gonna change 'timeplot' to handle 'varargin' better - with the struct approach. Let me know if you're fine with this.

Reply:
How would you feel if we made 'sig.timeplot' only handle name-value paired arguments, and if you want to feed in multiple signals, feeding those in as a signal array or cell array? 
e.g. `sig.timeplot('signals', {t,events,x,y}, 'mode' 1, 'twin', 10)`
I would be happy to do this and think this would make the code look nicer and could parse input args and signals more efficiently, but if you're against it let me know.

Reply:
That approach isn't very MATLAB-y.  I think it would be a good idea for functions that don't require any input args but for timeplot this change would add an extra arg that the user has to type with little added benefit.  It seems like you're solving for a problem that isn't there with this change.

Reply:
> That approach isn't very MATLAB-y.

Can you elaborate on this?

> It seems like you're solving for a problem that isn't there with this change.

My rationale was to re-factor the code to make it more elegant and future-proof. I think it's a nice approach for any functions which take name-value paired arguments. But that's fine if you're against it.

Reply:
Unlike Python where dictionaries are often used as input args, MATLAB primarily uses positional args.  Name-Value pairs are used to set optional parameters and come after the required positional ones.  Although MATLAB have grown more friendly towards Name-Value pairs (they've added more support for implementing them in user-defined functions), I'm not confident that they will fully adopt the Python system.  

Considering that calling timeplot with no args would be pointless, it seems less elegant to me that one has to call the function with an added parameter (the 'signals' input), particularly as MATLAB users don't expect to call functions without positional args (unless it takes no args at all).  We currently don't wrap signals in cells very often (in fact I think I've only ever done it once), so wrapping them only for this function also strikes me as inelegant (conversely on the rare occasion that the user does have them wrapped, they can call timeplot like this: `sig.timeplot(sigCell{:}, 'tWin', 20)`.  

To expand on the positional vs free point: the frequent use of dicts in Python means that it is very convenient to pass them in and out of functions, however in MATLAB there is no direct equivalent.  The closest thing to this is the struct which is rarely used as an input arg and again, only for optional ones.  Cell arrays could be used like a dict because the Name-Value pairs are 'packed' into a varargin array when they enter a function, but because they're not usually the output, they're not that convenient to pass around.